drivers/serial/Kconfig.bt

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2024 Croxel, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+config UART_BT
+	bool "UART over NUS Bluetooth LE"
+	depends on BT_NUS
+	depends on DT_HAS_ZEPHYR_NUS_UART_ENABLED
+	select UART_INTERRUPT_DRIVEN
+	select RING_BUFFER
+	select EXPERIMENTAL
+	help
+	  Enable the UART over NUS Bluetooth driver, which can be used to pipe
+	  serial data over Bluetooth LE GATT using NUS (Nordic UART Service).
+
+if UART_BT
+
+# Disable default NUS instance to not confuse it with the instances defined on UART_BT
+config BT_NUS_DEFAULT_INSTANCE
+	bool
+	default n
```

Comment:
minor: n is the "default default" so this line is redundant

Reply:
This only overrides the default value of `BT_NUS_DEFAULT_INSTANCE` to `n` when `UART_BT` is enabled. The actual Kconfig is defined under `subsys/bluetooth/services/nus/Kconfig`, and its purpose is to allow users to use NUS as a standalone GATT Service.

This means:
- When `BT_NUS` is enabled, but `UART_BT` is not, the service is instantiated within the `BT_NUS` module.
- When both `BT_NUS` and `UART_BT `are enabled, we don't have a default instance and we expect the user to define `nus-uart` DTS instances.

I'll add a more descriptive comment on top of this Kconfig symbol to reflect this purpose, but LMK if there's a better way to achieve this goal.

Reply:
Oh, I see. I haven't seen this kind of usage of Kconfig overlays elsewhere. Normally (AFAIK) those are used in board files and application overlays, but not in drivers or subsystems. Would it perhaps make sense to make the original definition conditional to `UART_BT`, i.e. instead of `default y` you'd have `default y if !UART_BT` there?

Reply:
That sounds good. I'll go ahead and make that change, instead.